### PR TITLE
must-gather: add more info for ceph crash

### PR DESCRIPTION
**Pull request by crombus**
add command to check for ceph crash info
for every crash and collect core dump of
rook crash.

Signed-off-by: crombus <pkundra@redhat.com>

### CODE REVIEW

File: must-gather/collection-scripts/gather_ceph_resources [hunk 1]
```diff
@@ -159,4 +162,15 @@ for ns in $namespaces; do
         { timeout 120 oc debug nodes/"${node}" -- bash -c "test -f /host/var/lib/rook/log/${ns}/ceph-volume.log && cat /host/var/lib/rook/log/${ns}/ceph-volume.log" > "${NODE_OUTPUT_DIR}"/ceph-volume.log; } >> "${BASE_COLLECTION_PATH}"/gather-debug.log 2>&1
     done
     oc delete -f pod_helper.yaml
+
+    # Collecting ceph prepare volume logs
```
Comment (ashishkumsingh) [inline must-gather/collection-scripts/gather_ceph_resources:166]: The comment needs to be changed to capturing crash dumps.

Comment (crombus) [reply must-gather/collection-scripts/gather_ceph_resources]: done

File: must-gather/collection-scripts/gather_ceph_resources [hunk 2]
```diff
@@ -159,4 +162,15 @@ for ns in $namespaces; do
         { timeout 120 oc debug nodes/"${node}" -- bash -c "test -f /host/var/lib/rook/log/${ns}/ceph-volume.log && cat /host/var/lib/rook/log/${ns}/ceph-volume.log" > "${NODE_OUTPUT_DIR}"/ceph-volume.log; } >> "${BASE_COLLECTION_PATH}"/gather-debug.log 2>&1
     done
     oc delete -f pod_helper.yaml
+
+    # Collecting ceph crash dump
+    for node in $(oc get nodes -l cluster.ocs.openshift.io/openshift-storage='' --no-headers | grep -w 'Ready' | awk '{print $1}'); do
```
Comment (leseb) [inline must-gather/collection-scripts/gather_ceph_resources:167]:
```suggestion
    for node in $(oc get nodes -l cluster.ocs.openshift.io/openshift-storage='' --no-headers | awk '/Ready/ {print $1}'); do
```

File: must-gather/collection-scripts/gather_ceph_resources [hunk 3]
```diff
@@ -159,4 +162,15 @@ for ns in $namespaces; do
...
+        printf "collecting crash logs  from node %s \n"  "${node}" | tee -a  "${BASE_COLLECTION_PATH}"/gather-debug.log
+        CRASH_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/crash_${node}
+        mkdir -p "${CRASH_OUTPUT_DIR}"
+        oc debug nodes/"${node}"  --to-namespace="${ns}" -- bash -c "sleep 5m" &  >> "${BASE_COLLECTION_PATH}"/gather-debug.log 2>&1
```
Comment (leseb) [inline must-gather/collection-scripts/gather_ceph_resources:171]:
```suggestion
        oc debug nodes/"${node}" --to-namespace="${ns}" -- bash -c "sleep 5m" &  >> "${BASE_COLLECTION_PATH}"/gather-debug.log 2>&1
```

File: must-gather/collection-scripts/gather_ceph_resources [hunk 4]
```diff
@@ -159,4 +162,15 @@ for ns in $namespaces; do
...
+        sleep 60
+        oc rsync -n "${ns}" `oc get pods -n "${ns}"| grep debug| awk '{print $1}'`:/host/var/lib/rook/openshift-storage/crash/ "${CRASH_OUTPUT_DIR}"
```
Comment (leseb) [inline must-gather/collection-scripts/gather_ceph_resources:173]:
```suggestion
        oc rsync -n "${ns}" $(oc get pods -n "${ns}"|awk '/debug/ {print $1}'):/host/var/lib/rook/openshift-storage/crash/ "${CRASH_OUTPUT_DIR}"
```

File: must-gather/collection-scripts/gather_ceph_resources [hunk 5]
```diff
@@ -136,6 +136,9 @@ for ns in $namespaces; do
         for i in $(timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "ceph osd lspools --connect-timeout=15"|awk '{print $2}'); do
             { timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "rbd ls -p $i" >> "${COMMAND_OUTPUT_DIR}/pools_rbd_$i"; } >> "${BASE_COLLECTION_PATH}"/gather-debug.log 2>&1;
         done
+        for i in $(timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "ceph crash ls --connect-timeout=15"|awk '{print $1}'); do
+            { timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "ceph crash info $i" >> "${COMMAND_OUTPUT_DIR}/pools_rbd_$i"; } >> "${BASE_COLLECTION_PATH}"/gather-debug.log 2>&1;
```
Comment (leseb) [inline must-gather/collection-scripts/gather_ceph_resources:140]: don't you need `--connect-timeout=15` here as well?

File: must-gather/collection-scripts/gather_ceph_resources (hunk 2 above)
Comment (crombus) [reply must-gather/collection-scripts/gather_ceph_resources]: done

File: must-gather/collection-scripts/gather_ceph_resources (hunk 3 above)
Comment (crombus) [reply must-gather/collection-scripts/gather_ceph_resources]: done

File: must-gather/collection-scripts/gather_ceph_resources (hunk 4 above)
Comment (crombus) [reply must-gather/collection-scripts/gather_ceph_resources]: done

File: must-gather/collection-scripts/gather_ceph_resources (hunk 5 above)
Comment (crombus) [reply must-gather/collection-scripts/gather_ceph_resources]: added

File: must-gather/collection-scripts/gather_ceph_resources [hunk 6]
```diff
@@ -159,4 +162,15 @@ for ns in $namespaces; do
...
     oc delete -f pod_helper.yaml
+
+    # Collecting ceph crash dump
+    for node in $(oc get nodes -l cluster.ocs.openshift.io/openshift-storage='' --no-headers | awk '/Ready/{print $1}'); do
+        printf "collecting crash logs  from node %s \n"  "${node}" | tee -a  "${BASE_COLLECTION_PATH}"/gather-debug.log
+        CRASH_OUTPUT_DIR=${CEPH_COLLECTION_PATH}/crash_${node}
+        mkdir -p "${CRASH_OUTPUT_DIR}"
+        oc debug nodes/"${node}" --to-namespace="${ns}" -- bash -c "sleep 5m" &  >> "${BASE_COLLECTION_PATH}"/gather-debug.log 2>&1
+        sleep 60
+        oc rsync -n "${ns}" `oc get pods -n "${ns}"| awk '/debug/{print $1}'`:/host/var/lib/rook/openshift-storage/crash/ "${CRASH_OUTPUT_DIR}"
```
Comment (leseb) [inline must-gather/collection-scripts/gather_ceph_resources:173]: backticks are deprecated use `$()` instead

Comment (crombus) [reply must-gather/collection-scripts/gather_ceph_resources]: done

**Review by jarrpa — APPROVED** [review]

File: must-gather/collection-scripts/gather_ceph_resources [hunk 7]
```diff
@@ -136,6 +136,7 @@ for ns in $namespaces; do
         for i in $(timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "ceph osd lspools --connect-timeout=15"|awk '{print $2}'); do
             { timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "rbd ls -p $i" >> "${COMMAND_OUTPUT_DIR}/pools_rbd_$i"; } >> "${BASE_COLLECTION_PATH}"/gather-debug.log 2>&1;
         done
+<<<<<<< HEAD
```
Comment (jarrpa) [inline must-gather/collection-scripts/gather_ceph_resources:139]: Remove this.

Comment (crombus) [reply must-gather/collection-scripts/gather_ceph_resources]: done

**Review by agarwal-mudit — APPROVED** [review]
lgtm

**Pull request merged**